libs/wingsdk/src/expect/assert.ts

```diff
@@ -62,5 +62,120 @@ export class Util {
     );
   }
 
+  /**
+   * Checks if a string matches a regular expression pattern.
+   * @param actual The string to test.
+   * @param expected The regular expression pattern to match against.
+   * @throws Will throw an error if the actual value does not match the expected regular expression pattern.
+   * @returns void
+   */
+  public static match(actual: string, expected: any): void {
+    const regex = new RegExp(expected);
+    nodeAssert.match(actual, regex);
+  }
+
+  /**
+   * Checks if a string does not match a regular expression pattern.
+   * @param actual The string to test.
+   * @param expected The regular expression pattern to check against.
+   * @throws Will throw an error if the actual value matches the expected regular expression pattern.
+   * @returns void
+   */
+  public static doesNotMatch(actual: string, expected: any): void {
+    const regex = new RegExp(expected);
+    nodeAssert.doesNotMatch(actual, regex);
+  }
+  /**
+   * Checks if a function throws an error with a specific message.
+   * @param fn The function to test.
+   * @throws Will throw an error if the provided function does not throw an error .
+   * @returns void
+   */
+  public static throws(fn: any): void {
+    nodeAssert.throws(fn);
+  }
+
+  /**
+   * Asserts that a function does not throw an error.
+   * @param fn The function to test.
+   * @throws Will throw an error if the function `fn` throws an error.
+   * @returns void
+   */
+  public static doesNotThrow(fn: any): void {
+    nodeAssert.doesNotThrow(fn);
+  }
+
+  /**
+   * Marks a test as failed.
+   * @param message An optional message to include with the failure.
+   * @throws Always throws an error with the provided message.
+   * @returns void
+   */
+  public static fail(message?: string): void {
+    nodeAssert.fail(message);
+  }
+
+  /**
+   * Asserts that a condition is truthy.
+   * @param condition The condition to test.
+   * @param message An optional message to include if the condition is falsy.
+   * @throws Will throw an error if the condition is falsy.
+   * @returns void
+   */
+  public static ok(condition: boolean, message?: string): void {
+    nodeAssert.ok(condition, message);
+  }
+
+  /**
+   * Checks if a value contains another value.
+   * Supports arrays, sets, and strings.
+   * @param actual The value to test.
+   * @param expected The value to check for containment.
+   * @throws Will throw an error if the actual value does not contain the expected value.
+   * @returns void
+   */
+  public static contains<T>(actual: unknown, expected: T | string): void {
```

Comment:
The signature of this method might cause issues in Wing since the language doesn't support type parameters yet, and we don't want the method to be typed with `any` parameters.

I think we should hold off on adding the `contains()` and `doesNotContain()` methods until we have some way to model iterator / iterable types - see https://github.com/winglang/wing/issues/4036

In the meantime it should be possible to work around this by using another assertion:

```js
// main.w
bring expect;

let my_array = [3, 4, 5];
expect.ok(my_array.contains(6), "missing element 6");
```